utilities/transactions/write_prepared_txn.cc

```diff
@@ -312,10 +312,9 @@ Status WritePreparedTxn::RollbackInternal() {
   const bool DISABLE_MEMTABLE = true;
   const uint64_t NO_REF_LOG = 0;
   uint64_t seq_used = kMaxSequenceNumber;
-  const size_t ZERO_PREPARES = 0;
   const size_t ONE_BATCH = 1;
   WritePreparedCommitEntryPreReleaseCallback update_commit_map(
-      wpt_db_, db_impl_, kMaxSequenceNumber, ZERO_PREPARES, ONE_BATCH);
+      wpt_db_, db_impl_, GetId(), prepare_batch_cnt_, ONE_BATCH);
```

Comment:
oops, so rollback was not handling duplicated keys?

Reply:
if that was the case, we should have passed ONE_PREPARES not ZERO_PREPARES. Zero prepares here means that we do not ask PreReleaseCallback to commit any prepared records. The idea was that since the transaction is aborted it naturally should not be committed.
The counter-intuitive change here is to commit the aborted transaction instead. Refer to the RP description for more details.

Reply:
You can describe what are the entries you add to commit map as inline comment.